crates/control/src/rule_obstacle_composer.rs

```diff
@@ -75,7 +76,7 @@ impl RuleObstacleComposer {
                 _,
             ) => {
                 let center_circle_obstacle = RuleObstacle::Circle(Circle::new(
-                    Point2::origin(),
+                    Framed::origin(),
```

Comment:
Point ... I think you should search for all occurrences of `Framed` and try to replace them if appropriate

Reply:
Done, I checked all references to the `origin` and `zeros` constructors.